Sources/Segment/Utilities/Utils.swift

```diff
@@ -0,0 +1,26 @@
+//
+//  Utils.swift
+//  Segment
+//
+//  Created by Brandon Sneed on 5/18/21.
+//
+
+import Foundation
+
+internal var isUnitTesting: Bool = {
+    let env = ProcessInfo().environment
+    let value = (env["XCTestConfigurationFilePath"] != nil)
+    return value
+}()
+
+internal let _SegmentException = "SegmentException"
+internal func exceptionFailure(_ message: String) {
+    let args: [CVarArg] = []
+    if isUnitTesting {
+        NSException.raise(NSExceptionName(_SegmentException), format: message, arguments: getVaList(args))
```

Comment:
Are we sure we want to use NSExceptions? They aren't friendly with recovery

Reply:
Yeah, that's why I used them.  I want a hard stop when unit testing or during development.

Reply:
Awesome, if that is the intention then we good :)